### PR TITLE
Display a notice when there are plugins installed or updates available

### PR DESCRIPTION
**Pull request by janfaracik**
Followup to #8186. Adds a notice to the Installed plugins page(s) as well as to the updates page to tell the user when there are no plugins installed/updates available, this is rather than displaying the table. Also disables the search bar.

image

image

image

### Testing done

* Notice shows when no plugins are installed
* Notice hides when plugins are installed
* Notice shows when no updates are available
* Notice hides when updates are available

### Proposed changelog entries

- Display a notice when plugin updates are available or when there are no plugins installed.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@jenkinsci/sig-ux 

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8208"></a>

### CODE REVIEW

File: core/src/main/resources/hudson/PluginManager/updates.jelly
Comment (NotMyFault) [inline core/src/main/resources/hudson/PluginManager/updates.jelly:228]:
```suggestion
            <l:notice icon="symbol-up-to-date" text="${%No updates available}" />
```
Given the updated presence, this sounds more fitting to me, thoughts?

**Review by NotMyFault — COMMENTED** [review]
Tests will need a few fixes too, as it seems

File: core/src/main/resources/hudson/PluginManager/updates.jelly
Comment (janfaracik) [reply core/src/main/resources/hudson/PluginManager/updates.jelly:228]: I like it!

**Review by yaroslavafenkin — COMMENTED** [review]
Looks fine security wise

**Comment by janfaracik** [comment]
I've updated this to include the Installed plugins page(s) which had slipped my mind before, and I've also removed the usage of `:has()` as it was breaking the tests ([for the time being](https://github.com/HtmlUnit/htmlunit-cssparser/issues/20)). Rereviews and another security look over would be appreciated. Thanks

**Comment by daniel-beck** [comment]
For UI change PRs in general, it would be useful to have both "before" and "after" screenshots (not just "after"). For example, <details><summary>this</summary>

Screenshot 2023-07-27 at 10 46 12

</details> is what the plugin manager currently looks like. While the suggested change does look better, the title and description of this PR are misleading. In all three cases, a message is already shown, there's not just a table header.

**Review by timja — APPROVED** [review]
Looks really nice

**Review by NotMyFault — APPROVED** [review]
This looks nice!

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
